05-html-css/css-framework.md

```diff
@@ -18,185 +16,144 @@ The usual components are:
   <a href="#">There is an error here!</a>
 </section>
 ```
 
-![alert callout screenshot](imgs/alert-calloutv2.png)
-
-You can experiment with callouts [here](https://codepen.io/adadev/pen/dmQodr?editors=1000).
-
-**Experiment**:  Change the color of the callout background.
-
-## Form Validations
-
-Foundation also has a library named [**Abide**](https://foundation.zurb.com/sites/docs/abide.html) which can be used to go beyond the built-in HTML5 form validation available to us.  To use Abide over HTML5 validations, add `novalidate` and `data-abide` attributes to the form element.  
-
-Elements can have a number of validations which ensure input matches common patterns like `alpha` (only letters), `alpha_numeric`, `color`, `cvv` (the three numbers on the back of a credit card), `date`, `datetime`, `day_month_year`, `domain`, `email`, `integer`, `month_day_year`, `number`, `time`, and `url`.  
+![alert callout screenshot](imgs/bootstrap-callout.png)
 
-A pattern can be specified by adding a pattern attribute-value to the form field.  For example `<input type="text" pattern="email" >`.  Then when the user attempts to submit the form, Abide will check to ensure the field matches the given pattern.  
+You can experiment with alerts [here](https://codepen.io/adadev/pen/VEjKqz?editors=1000).
 
-Developers can also add an attribute `required` to any form field which is required to have a value.  
+**Experiment**:  Change the color of the alert background.
 
-Abide can also use a `form-error` field to display an error message to the user when a field fails to validate.  To use a `form-error`, group labels, inputs and a span or other text field with the class `form-error` together in a container.  The text with the class `form-error` will display in red if and only if the form is submitted with invalid data in the accompanying field.  You can experiment with a code example [here](https://codepen.io/adadev/pen/vRQKoZ?editors=1100).
-
-```html
-<form data-abide novalidate>
-  <p>This example shows input validation using Abide.</p>
-  <div>
-      <input type="text" required>
-       <span class="form-error">Please enter amount.</span>
-  </div>
-  <div>
-      <input type="text" placeholder="www.somesite.com" required pattern="url">
-      <span class="form-error">Please enter a website URL.</span>  
-  </div>    
-
-  <button type="submit" class="button" value="Submit">Submit to see Errors</button>
-</form>
-```
-
-**Question:** If I have validations in my HTML with Foundation, do I need model validations on the back-end?
 
 ## Best Practices
-- NEVER change any CSS in Foundation's documents
-  - It's _okay_ to override them, if necessary, in your css files or the `_settings.scss` file.  You can use the `foundation_and_overrides.scss` file to turn off or on sections of the Foundation library.
+- NEVER change any CSS in Bootstrap's documents
+  - It's _okay_ to override them, if necessary, in your css files.
```

Comment:
I know this was already here, but I'm not sure it's clear what "override them in your css files" means. Could we provide an example of that in the context of Rails (not a codepen)?

Reply:
I added a section earlier on about overriding.